schema.go

```diff
@@ -48,14 +49,37 @@ func (s *Schema[T]) Refine(predicate func(T) bool) *Schema[T] {
 	return s
 }
 
+func (s *Schema[T]) _IsOptional() bool {
+	if s.isOptional == nil {
+		return false
+	}
+	return *s.isOptional
+}
```

Comment:
```suggestion
```

We can just reference `s.isOptional` property on the schema type if we make it not nullable.